### PR TITLE
TS-3746: make proxy.config.ssl.client.verify.server overridable

### PR DESCRIPTION
**Pull request by persiaAziz**

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci autest]

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -5491,12 +5491,19 @@ HttpSM::handle_http_server_open()
...
-                          vc->options.packet_tos != t_state.txn_conf->sock_packet_tos_out)) {
-      vc->options.sockopt_flags = t_state.txn_conf->sock_option_flag_out;
-      vc->options.packet_mark   = t_state.txn_conf->sock_packet_mark_out;
-      vc->options.packet_tos    = t_state.txn_conf->sock_packet_tos_out;
+
+    //    SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(vc);
+    //    if (ssl_vc) {
+    //      ssl_vc->setClientVerifyEnable(t_state.txn_conf->ssl_client_verify_server);
+    //    }
+    if (vc != NULL && (vc->options.sockopt_flags != t_state.txn_conf->sock_option_flag_out ||
```
Comment (zwoop) [inline proxy/http/HttpSM.cc]: This ought to be nullptr still no?

Comment (persiaAziz) [inline proxy/http/HttpSM.cc]: yes it should be nullptr. bad cherry-pick

**Comment by persiaAziz** [comment]
[approve ci autest]

**Pull request merged**